### PR TITLE
Code to review

### PR DESCRIPTION
**Pull request by seperalta22**
# Excercise: Does my coding partner follow JS best practices?

This branch contains a copy of my ToDo List project so that it can be reviewed by my coding partner.

### CODE REVIEW

File: src/modules/addList.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  }
+
+  const arrayTask = JSON.parse(localStorage.getItem('tasks'));
+  for (let i = 0; i < arrayTask.length; i += 1) {
+    for (let j = 0; j < arrayTask.length; j += 1) {
+      if (arrayTask[j].index === i + 1) {
+        index = j;
+        break;
+      }
+    }
```
Comment (joramot) [inline src/modules/addList.js:24]: Kindly suggest replaced by a simpler array method, such as find() or filter().

File: src/modules/addList.js
Comment (joramot) [inline src/modules/addList.js:1]: Some areas that could be refactored to reduce code duplication. For example, the logic to update the appearance of a task item when it is completed or unchecked is repeated in various parts of the code and could be refactored into a separate function.

**Review by joramot — COMMENTED** [review]
Hello @SergioPeralta22 You have done an excellent job and have applied the best practices KISS, DRY and JavaScript best practices., DRY, KISS, and YAGNI rules and HTML & CSS best practices.

- Kindly you could consider adding validations and error handling when saving data to local storage to avoid unexpected failures.
- You might consider using a mechanism to ensure that data saved to local storage is of the expected type when retrieved.
- It would be advisable to add comments to make it easier for other developers to understand.

**Comment by seperalta22** [comment]
@joramot Thank you very much for your advice

**Pull request merged**